### PR TITLE
Add QoS parameter for MQTT Publish

### PR DESCRIPTION
**Pull request by jbouwh**
## Proposed change

Add QoS parameter for MQTT Publish. In the current situation always QoS `0` is used. This PR allows the `QoS` to be set when publishing an MQTT payload.
Adding an QoS parameter for testing MQTT subscribe is planned as well.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

File: src/panels/config/integrations/integration-panels/mqtt/mqtt-config-panel.ts [hunk 1]
```diff
@@ -54,6 +60,18 @@ class HaPanelDevMqtt extends LitElement {
...
                 @change=${this._handleTopic}
               ></ha-textfield>
+              <ha-select
+                .label=${this.hass.localize("ui.panel.config.mqtt.qos")}
+                .value=${this.qos}
+                @selected=${this._handleQos}
+                @closed=${stopPropagation}
+                fixedMenuPosition
+                naturalMenuWidth
+                >${["0", "1", "2"].map(
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/mqtt/mqtt-config-panel.ts:70]: make this array a `const` outside the class, so it is not recreated on every render.

File: src/panels/config/integrations/integration-panels/mqtt/mqtt-config-panel.ts [hunk 2]
```diff
@@ -95,13 +113,22 @@ class HaPanelDevMqtt extends LitElement {
     }
   }
 
+  private _handleQos(ev: CustomEvent) {
+    const newValue = (ev.target! as any).value;
+    if (newValue >= 0 && newValue !== this.qos && localStorage && this.inited) {
+      this.qos = newValue;
+      localStorage["panel-dev-mqtt-qos"] = this.qos;
```
Comment (bramkragten) [inline src/panels/config/integrations/integration-panels/mqtt/mqtt-config-panel.ts:120]: We should use the `@LocalStorage` decorator :-) (doesn't have to be in this PR)

Comment (jbouwh) [reply src/panels/config/integrations/integration-panels/mqtt/mqtt-config-panel.ts:118]: Ah, good follow up PR then.

**Review by bramkragten — APPROVED** [review]

**Comment by jbouwh** [comment]
@bramkragten #14565 implements the MQTT subscribe part of the panel. The backend change for this change has been merged.

**Comment by jbouwh** [comment]
@bramkragten Can this be merged, or am I missing something?

**Pull request merged**